[SwiftUI] Update WebPage to latest interface (Part 1)

rr-codes opened the pull request:
#### 42e0d3664d57c157f406d25b1e86ff5edcb80834
<pre>
[SwiftUI] Update WebPage to latest interface (Part 1)
<a href="https://bugs.webkit.org/show_bug.cgi?id=286823">https://bugs.webkit.org/show_bug.cgi?id=286823</a>
<a href="https://rdar.apple.com/143977907">rdar://143977907</a>

Reviewed by Aditya Keerthi.

Update various definitions/declarations.

* Source/WebKit/UIProcess/API/Swift/DownloadCoordinator.swift:
(DownloadCoordinator.destination(forDownload:response:suggestedFilename:)):
(DownloadCoordinator.authenticationChallengeDisposition(forDownload:challenge:URLCredential:)):
(DownloadCoordinator.httpRedirectionPolicy(forDownload:response:newRequest:)):
(DownloadCoordinator.placeholderPolicy(forDownload:)):
* Source/WebKit/UIProcess/API/Swift/WKDownloadDelegateAdapter.swift:
(WKDownloadDelegateAdapter.owner):
* Source/WebKit/UIProcess/API/Swift/WKNavigationDelegateAdapter.swift:
(WKNavigationDelegateAdapter.owner):
(WKNavigationDelegateAdapter.yieldNavigationProgress(_:cocoaNavigation:)):
(WKNavigationDelegateAdapter.yieldDownloadProgress(_:download:)):
(WKNavigationDelegateAdapter.webView(_:decidePolicyFor:preferences:WKWebpagePreferences:)):
(WKNavigationDelegateAdapter.webView(_:decidePolicyFor:)):
* Source/WebKit/UIProcess/API/Swift/WKUIDelegateAdapter.swift:
(WKUIDelegateAdapter.owner):
(WKUIDelegateAdapter.menuBuilder):
(WKUIDelegateAdapter._webView(_:getContextMenuFromProposedMenu:forElement:userInfo:)):
* Source/WebKit/UIProcess/API/Swift/WebPage+BackForwardList.swift:
* Source/WebKit/UIProcess/API/Swift/WebPage+Configuration.swift:
(Configuration.defaultNavigationPreferences):
(Configuration.deviceSensorAuthorization):
* Source/WebKit/UIProcess/API/Swift/WebPage+DialogPresenting.swift:
(DialogPresenting.handleJavaScriptAlert(_:initiatedBy:)):
(DialogPresenting.handleJavaScriptConfirm(_:initiatedBy:)):
(DialogPresenting.handleJavaScriptPrompt(_:defaultText:initiatedBy:)):
(DialogPresenting.handleFileInputPrompt(_:initiatedBy:)):
* Source/WebKit/UIProcess/API/Swift/WebPage+Download.swift:
(originatingFrame):
* Source/WebKit/UIProcess/API/Swift/WebPage+ElementInfo.swift:
* Source/WebKit/UIProcess/API/Swift/WebPage+FrameInfo.swift:
* Source/WebKit/UIProcess/API/Swift/WebPage+Navigation.swift:
(Navigations.makeAsyncIterator): Deleted.
(Iterator.source): Deleted.
(Iterator.next): Deleted.
* Source/WebKit/UIProcess/API/Swift/WebPage+NavigationDeciding.swift:
(NavigationDeciding.decidePolicy(for:preferences:)):
(NavigationDeciding.decidePolicy(for:)):
* Source/WebKit/UIProcess/API/Swift/WebPage+NavigationPreferences.swift:
* Source/WebKit/UIProcess/API/Swift/WebPage.swift:
(currentNavigationEvent):
(fullscreenState):
(mediaType):
(setMenuBuilder(_:)):
(load(_:baseURL:)):
(load(simulatedRequest:responseHTML:)):
(load(_:)):
(callJavaScript(_:arguments:in:contentWorld:)):
(startDownload(using:)):
(resumeDownload(fromResumeData:)):
(KeyValueObservations.contents):
(handlesURLScheme(_:)): Deleted.
(callAsyncJavaScript(_:arguments:in:contentWorld:)): Deleted.
(snapshot(_:)): Deleted.
* Source/WebKit/_WebKit_SwiftUI/View+WebViewModifiers.swift:
* Source/WebKit/_WebKit_SwiftUI/WebPage+SwiftUI.swift:
(WebPage.snapshot(_:)):
(WebPage_v0.themeColor): Deleted.
* Source/WebKit/_WebKit_SwiftUI/WebPageNavigationAction+SwiftUI.swift:
(WebPage_v0.modifierFlags): Deleted.
* Source/WebKit/_WebKit_SwiftUI/WebView.swift:
* Tools/SwiftBrowser/Source/ViewModel/BrowserViewModel.swift:
* Tools/SwiftBrowser/Source/ViewModel/DialogPresenter.swift:
(handleJavaScriptAlert(_:initiatedBy:)):
(handleJavaScriptConfirm(_:initiatedBy:)):
(handleJavaScriptPrompt(_:defaultText:initiatedBy:)):
(handleFileInputPrompt(_:initiatedBy:)):
* Tools/SwiftBrowser/Source/ViewModel/DownloadCoordinator.swift:
(DownloadCoordinator.didReceiveDownloadEvent(_:)):
(DownloadCoordinator.destination(forDownload:response:suggestedFilename:)):
* Tools/SwiftBrowser/Source/ViewModel/NavigationDecider.swift:
(NavigationDecider.decidePolicy(for:preferences:)):
(NavigationDecider.decidePolicy(for:)):
* Tools/SwiftBrowser/Source/Views/ContentView.swift:
(ContentView.body):
* Tools/TestWebKitAPI/Tests/WebKit Swift/URLSchemeHandlerTests.swift:
(URLSchemeHandlerTests.basicSchemeHandling):
* Tools/TestWebKitAPI/Tests/WebKit Swift/WebPageTests.swift:
(TestNavigationDecider.preferencesMutation):
(TestNavigationDecider.decidePolicy(for:preferences:)):
(TestNavigationDecider.decidePolicy(for:)):
(WebPageTests.observableProperties):
(WebPageTests.decidePolicyForNavigationActionFragment):
(WebPageTests.javaScriptEvaluation):
(WebPageTests.decidePolicyForNavigationResponse):
(WebPageTests.basicNavigation): Deleted.
(WebPageTests.sequenceOfNavigations): Deleted.
(WebPageTests.navigationWithFailedProvisionalNavigationEvent): Deleted.
(WebPageTests.navigationPreferencesMutationDuringNavigation): Deleted.

Canonical link: <a href="https://commits.webkit.org/290097@main">https://commits.webkit.org/290097@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/157290e6cdd42a3431ad602c1720e0459dc52107

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/88978 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8502 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/43577 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/93950 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39738 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91029 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8889 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16686 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/68550 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/26228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/91980 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/6783 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/80555 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/48914 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/6532 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/34966 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38846 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/76894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/35893 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95789 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16158 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/11801 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/77432 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16414 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/76366 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/76721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/21107 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/19611 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9241 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13939 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16172 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21483 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/15913 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19364 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17694 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebKit/UIProcess/API/Swift/WebPage+Navigation.swift:
```diff
@@ -34,19 +35,28 @@ extension WebPage_v0 {
+            /// This event occurs when the web page received a server redirect for a request.
             case receivedServerRedirect
 
+            /// This event occurs when the web page has started to receive content for the main frame.`
```
```suggestion
            /// This event occurs when the web page has started to receive content for the main frame.
```

pxlcoder commented on Source/WebKit/UIProcess/API/Swift/WebPage.swift:
```diff
@@ -151,98 +185,155 @@ public class WebPage_v0 {
     // MARK: Properties
 
-    public let navigations: Navigations
+    /// The current navigation event, or `nil` if there have been no navigations so far..
```
```suggestion
    /// The current navigation event, or `nil` if there have been no navigations so far.
```

pxlcoder commented on Source/WebKit/UIProcess/API/Swift/WebPage.swift:
```diff
@@ -151,98 +185,155 @@ public class WebPage_v0 {
 
+    /// The trust management object you use to evaluate trust for the current webpage.
+    ///
+    /// Use the object in this property to validate the webpage’s certificate and associated credentials.
```
`webpage` or `web page`?

pxlcoder commented on Source/WebKit/UIProcess/API/Swift/WebPage.swift:
```diff
@@ -151,98 +185,155 @@ public class WebPage_v0 {
         backingProperty(\.serverTrust, backedBy: \.serverTrust)
     }
 
+    /// Indicates whether the web page loaded all resources on the page through securely encrypted connections.
```
`webpage` or `web page`?

pxlcoder commented on Source/WebKit/UIProcess/API/Swift/WebPage+Navigation.swift:
```diff
@@ -56,48 +66,12 @@ extension WebPage_v0 {
+        /// The type of this navigation event.
         public let kind: Kind
 
+        /// The id of the navigation that triggered this event. Multiple sequential events will have the same navigation identifier.
```
```suggestion
        /// The ID of the navigation that triggered this event. Multiple sequential events will have the same navigation identifier.
```

pxlcoder commented on Source/WebKit/UIProcess/API/Swift/WebPage.swift:
```diff
@@ -280,11 +371,31 @@ public class WebPage_v0 {
+    ///
+    /// Provide the source of this load request for app activity data by setting the attribution parameter on your request.
+    ///
+    /// - Parameter request: A URL request that specifies the resource to display
```
```suggestion
    /// - Parameter request: A URL request that specifies the resource to display.
```

pxlcoder commented on Source/WebKit/_WebKit_SwiftUI/WebPage+SwiftUI.swift:
```diff
@@ -36,4 +37,18 @@ extension WebPage_v0 {
         }
     }
+
+    /// Generates an image from the web view’s contents
```
```suggestion
    /// Generates an image from the web view’s contents.
```

pxlcoder reviewed (approved):
I did not comment in every place, but both `web page` and `webpage` are used in inconsistently used in many documentation comments.

rr-codes committed:
[SwiftUI] Update WebPage to latest interface (Part 1)
https://bugs.webkit.org/show_bug.cgi?id=286823
rdar://143977907

Reviewed by Aditya Keerthi.

Update various definitions/declarations.

* Source/WebKit/UIProcess/API/Swift/DownloadCoordinator.swift:
(DownloadCoordinator.destination(forDownload:response:suggestedFilename:)):
(DownloadCoordinator.authenticationChallengeDisposition(forDownload:challenge:URLCredential:)):
(DownloadCoordinator.httpRedirectionPolicy(forDownload:response:newRequest:)):
(DownloadCoordinator.placeholderPolicy(forDownload:)):
* Source/WebKit/UIProcess/API/Swift/WKDownloadDelegateAdapter.swift:
(WKDownloadDelegateAdapter.owner):
* Source/WebKit/UIProcess/API/Swift/WKNavigationDelegateAdapter.swift:
(WKNavigationDelegateAdapter.owner):
(WKNavigationDelegateAdapter.yieldNavigationProgress(_:cocoaNavigation:)):
(WKNavigationDelegateAdapter.yieldDownloadProgress(_:download:)):
(WKNavigationDelegateAdapter.webView(_:decidePolicyFor:preferences:WKWebpagePreferences:)):
(WKNavigationDelegateAdapter.webView(_:decidePolicyFor:)):
* Source/WebKit/UIProcess/API/Swift/WKUIDelegateAdapter.swift:
(WKUIDelegateAdapter.owner):
(WKUIDelegateAdapter.menuBuilder):
(WKUIDelegateAdapter._webView(_:getContextMenuFromProposedMenu:forElement:userInfo:)):
* Source/WebKit/UIProcess/API/Swift/WebPage+BackForwardList.swift:
* Source/WebKit/UIProcess/API/Swift/WebPage+Configuration.swift:
(Configuration.defaultNavigationPreferences):
(Configuration.deviceSensorAuthorization):
* Source/WebKit/UIProcess/API/Swift/WebPage+DialogPresenting.swift:
(DialogPresenting.handleJavaScriptAlert(_:initiatedBy:)):
(DialogPresenting.handleJavaScriptConfirm(_:initiatedBy:)):
(DialogPresenting.handleJavaScriptPrompt(_:defaultText:initiatedBy:)):
(DialogPresenting.handleFileInputPrompt(_:initiatedBy:)):
* Source/WebKit/UIProcess/API/Swift/WebPage+Download.swift:
(originatingFrame):
* Source/WebKit/UIProcess/API/Swift/WebPage+ElementInfo.swift:
* Source/WebKit/UIProcess/API/Swift/WebPage+FrameInfo.swift:
* Source/WebKit/UIProcess/API/Swift/WebPage+Navigation.swift:
(Navigations.makeAsyncIterator): Deleted.
(Iterator.source): Deleted.
(Iterator.next): Deleted.
* Source/WebKit/UIProcess/API/Swift/WebPage+NavigationDeciding.swift:
(NavigationDeciding.decidePolicy(for:preferences:)):
(NavigationDeciding.decidePolicy(for:)):
* Source/WebKit/UIProcess/API/Swift/WebPage+NavigationPreferences.swift:
* Source/WebKit/UIProcess/API/Swift/WebPage.swift:
(currentNavigationEvent):
(fullscreenState):
(mediaType):
(setMenuBuilder(_:)):
(load(_:baseURL:)):
(load(simulatedRequest:responseHTML:)):
(load(_:)):
(callJavaScript(_:arguments:in:contentWorld:)):
(startDownload(using:)):
(resumeDownload(fromResumeData:)):
(KeyValueObservations.contents):
(handlesURLScheme(_:)): Deleted.
(callAsyncJavaScript(_:arguments:in:contentWorld:)): Deleted.
(snapshot(_:)): Deleted.
* Source/WebKit/_WebKit_SwiftUI/View+WebViewModifiers.swift:
* Source/WebKit/_WebKit_SwiftUI/WebPage+SwiftUI.swift:
(WebPage.snapshot(_:)):
(WebPage_v0.themeColor): Deleted.
* Source/WebKit/_WebKit_SwiftUI/WebPageNavigationAction+SwiftUI.swift:
(WebPage_v0.modifierFlags): Deleted.
* Source/WebKit/_WebKit_SwiftUI/WebView.swift:
* Tools/SwiftBrowser/Source/ViewModel/BrowserViewModel.swift:
* Tools/SwiftBrowser/Source/ViewModel/DialogPresenter.swift:
(handleJavaScriptAlert(_:initiatedBy:)):
(handleJavaScriptConfirm(_:initiatedBy:)):
(handleJavaScriptPrompt(_:defaultText:initiatedBy:)):
(handleFileInputPrompt(_:initiatedBy:)):
* Tools/SwiftBrowser/Source/ViewModel/DownloadCoordinator.swift:
(DownloadCoordinator.didReceiveDownloadEvent(_:)):
(DownloadCoordinator.destination(forDownload:response:suggestedFilename:)):
* Tools/SwiftBrowser/Source/ViewModel/NavigationDecider.swift:
(NavigationDecider.decidePolicy(for:preferences:)):
(NavigationDecider.decidePolicy(for:)):
* Tools/SwiftBrowser/Source/Views/ContentView.swift:
(ContentView.body):
* Tools/TestWebKitAPI/Tests/WebKit Swift/URLSchemeHandlerTests.swift:
(URLSchemeHandlerTests.basicSchemeHandling):
* Tools/TestWebKitAPI/Tests/WebKit Swift/WebPageTests.swift:
(TestNavigationDecider.preferencesMutation):
(TestNavigationDecider.decidePolicy(for:preferences:)):
(TestNavigationDecider.decidePolicy(for:)):
(WebPageTests.observableProperties):
(WebPageTests.decidePolicyForNavigationActionFragment):
(WebPageTests.javaScriptEvaluation):
(WebPageTests.decidePolicyForNavigationResponse):
(WebPageTests.basicNavigation): Deleted.
(WebPageTests.sequenceOfNavigations): Deleted.
(WebPageTests.navigationWithFailedProvisionalNavigationEvent): Deleted.
(WebPageTests.navigationPreferencesMutationDuringNavigation): Deleted.

Canonical link: https://commits.webkit.org/290097@main

axiixc commented on Source/WebKit/UIProcess/API/Swift/WebPage+Download.swift:
```diff
@@ -25,8 +25,9 @@
+extension WebPage {
     @MainActor
+    @_spi(Private)
     public struct Download: Sendable {
```
Why is this `@MainActor` and `Sendable`?

rr-codes replied:
ah true, I don't think it needs the `@MainActor` . Thanks!!

axiixc replied:
I think there were a couple of other types with both annotations, might be worth a quick audit. Technically, if a type is bound to a global actor it is _also_ Sendable, because the concurrency checker knows you won’t access it improperly. But writing both is a bit confusing w.r.t. intent. I think having a type that’s sendable but with one or two `@MainActor` vars or funds could make sense, but is probably rare.

Edit: Actually, [the docs](https://developer.apple.com/documentation/swift/sendable#Sendable-Classes) say this for classes but not structs… so maybe double check me on the specifics then. I’d assume it’s the same though?
